[Temporal] Add PlainMonthDay

catamorphism opened the pull request:
#### dab2cf707db19ca3bc7f2379b7217a9a47510506
<pre>
[Temporal] Add PlainMonthDay
<a href="https://bugs.webkit.org/show_bug.cgi?id=300389">https://bugs.webkit.org/show_bug.cgi?id=300389</a>

Reviewed by Yusuke Suzuki.

Add basic support for Temporal PlainMonthDay type, without most
methods yet.

Test: JSTests/stress/temporal-plainmonthday.js
* JSTests/stress/temporal-plainmonthday.js: Added.
(shouldBe):
(shouldThrow):
(const.monthDay.new.Temporal.PlainMonthDay):
* JSTests/test262/config.yaml:
* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/DerivedSources.make:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/runtime/ISO8601.h:
* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::init):
(JSC::JSGlobalObject::visitChildrenImpl):
* Source/JavaScriptCore/runtime/JSGlobalObject.h:
(JSC::JSGlobalObject::plainMonthDayStructure):
* Source/JavaScriptCore/runtime/TemporalObject.cpp:
(JSC::createPlainMonthDayConstructor):
(JSC::toTemporalOverflow):
* Source/JavaScriptCore/runtime/TemporalObject.h:
* Source/JavaScriptCore/runtime/TemporalPlainMonthDay.cpp: Added.
(JSC::TemporalPlainMonthDay::create):
(JSC::TemporalPlainMonthDay::createStructure):
(JSC::TemporalPlainMonthDay::TemporalPlainMonthDay):
(JSC::TemporalPlainMonthDay::finishCreation):
(JSC::TemporalPlainMonthDay::visitChildrenImpl):
(JSC::TemporalPlainMonthDay::tryCreateIfValid):
(JSC::TemporalPlainMonthDay::monthCode const):
* Source/JavaScriptCore/runtime/TemporalPlainMonthDay.h: Added.
* Source/JavaScriptCore/runtime/TemporalPlainMonthDayConstructor.cpp: Added.
(JSC::TemporalPlainMonthDayConstructor::create):
(JSC::TemporalPlainMonthDayConstructor::createStructure):
(JSC::TemporalPlainMonthDayConstructor::TemporalPlainMonthDayConstructor):
(JSC::TemporalPlainMonthDayConstructor::finishCreation):
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/runtime/TemporalPlainMonthDayConstructor.h: Added.
* Source/JavaScriptCore/runtime/TemporalPlainMonthDayPrototype.cpp: Added.
(JSC::TemporalPlainMonthDayPrototype::create):
(JSC::TemporalPlainMonthDayPrototype::createStructure):
(JSC::TemporalPlainMonthDayPrototype::TemporalPlainMonthDayPrototype):
(JSC::TemporalPlainMonthDayPrototype::finishCreation):
(JSC::JSC_DEFINE_CUSTOM_GETTER):
* Source/JavaScriptCore/runtime/TemporalPlainMonthDayPrototype.h: Added.

Canonical link: <a href="https://commits.webkit.org/301244@main">https://commits.webkit.org/301244@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/299221ff0f406bc53ef13f7835eaa67d374b1d6c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125293 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44958 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35699 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132143 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77161 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45646 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53525 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95395 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63327 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128247 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36451 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112044 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75936 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35350 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30213 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75619 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/117384 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106218 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30439 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134827 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/123811 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52097 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39878 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103865 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52535 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108266 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103626 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48987 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27277 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/49202 "The change is no longer eligible for processing.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19629 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51992 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57772 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/156830 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51351 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39271 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54706 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53044 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ea57bb20e69af6fae62c4b51301a7ef9f0ba15bf. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/52010

catamorphism committed:
[Temporal] Add PlainMonthDay
https://bugs.webkit.org/show_bug.cgi?id=300389

Reviewed by Yusuke Suzuki.

Add basic support for Temporal PlainMonthDay type, without most
methods yet.

Test: JSTests/stress/temporal-plainmonthday.js
* JSTests/stress/temporal-plainmonthday.js: Added.
(shouldBe):
(shouldThrow):
(const.monthDay.new.Temporal.PlainMonthDay):
* JSTests/test262/config.yaml:
* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/DerivedSources.make:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/runtime/ISO8601.h:
* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::init):
(JSC::JSGlobalObject::visitChildrenImpl):
* Source/JavaScriptCore/runtime/JSGlobalObject.h:
(JSC::JSGlobalObject::plainMonthDayStructure):
* Source/JavaScriptCore/runtime/TemporalObject.cpp:
(JSC::createPlainMonthDayConstructor):
(JSC::toTemporalOverflow):
* Source/JavaScriptCore/runtime/TemporalObject.h:
* Source/JavaScriptCore/runtime/TemporalPlainMonthDay.cpp: Added.
(JSC::TemporalPlainMonthDay::create):
(JSC::TemporalPlainMonthDay::createStructure):
(JSC::TemporalPlainMonthDay::TemporalPlainMonthDay):
(JSC::TemporalPlainMonthDay::finishCreation):
(JSC::TemporalPlainMonthDay::visitChildrenImpl):
(JSC::TemporalPlainMonthDay::tryCreateIfValid):
(JSC::TemporalPlainMonthDay::monthCode const):
* Source/JavaScriptCore/runtime/TemporalPlainMonthDay.h: Added.
* Source/JavaScriptCore/runtime/TemporalPlainMonthDayConstructor.cpp: Added.
(JSC::TemporalPlainMonthDayConstructor::create):
(JSC::TemporalPlainMonthDayConstructor::createStructure):
(JSC::TemporalPlainMonthDayConstructor::TemporalPlainMonthDayConstructor):
(JSC::TemporalPlainMonthDayConstructor::finishCreation):
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/runtime/TemporalPlainMonthDayConstructor.h: Added.
* Source/JavaScriptCore/runtime/TemporalPlainMonthDayPrototype.cpp: Added.
(JSC::TemporalPlainMonthDayPrototype::create):
(JSC::TemporalPlainMonthDayPrototype::createStructure):
(JSC::TemporalPlainMonthDayPrototype::TemporalPlainMonthDayPrototype):
(JSC::TemporalPlainMonthDayPrototype::finishCreation):
(JSC::JSC_DEFINE_CUSTOM_GETTER):
* Source/JavaScriptCore/runtime/TemporalPlainMonthDayPrototype.h: Added.

Canonical link: https://commits.webkit.org/301244@main
